ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryExecutor.java

```diff
@@ -229,7 +229,9 @@ public PersistentQueryMetadata buildPersistentQuery(
         ksqlConfig.getLong(KSQL_SHUTDOWN_TIMEOUT_MS_CONFIG),
         classifier,
         physicalPlan,
-        ksqlConfig.getInt(KsqlConfig.KSQL_QUERY_ERROR_MAX_QUEUE_SIZE)
+        ksqlConfig.getInt(KsqlConfig.KSQL_QUERY_ERROR_MAX_QUEUE_SIZE),
+        (ProcessingLogger) streamsProperties.get(
```

Comment:
We should use a different logger here. The logger is important because its name tells you a lot about the context of the error. In this case, the logger's name is `ksql.logger.production.error`, which tells you this is an error hit while trying to produce a record to Kafka. Other loggers' names include the node in the query topology. Eventually we'll let users adjust the log levels for some loggers at runtime, so you could see very detailed logging from just 1 UDF for example. In this case I'd name this something like `ksql.logger.thread.exception.uncaught`.

To create the logger, you can use the factory in the `processingLogContext` member of this class.

Reply:
Done